src/models/Airline.ts

```diff
@@ -0,0 +1,357 @@
+                    && aircraft.barometricAltitude < PlaneAlert.config.thresholds.landing
+                    && !this.aircraft[i].meta.onGround) {
+                    PlaneAlert.log.info(`Plane ${this.aircraft[i].icao} is landing`);
+                    //Plane landing
+                    const nearestAirport = this.findNearestAirport(this.aircraft[i]);
+                    if (nearestAirport !== null) {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) landed at ${nearestAirport.airport.name} (${nearestAirport.airport.icao})`);
+                    } else {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) landed`);
+                    }
+                    this.triggerEvent(PlaneEvents.PLANE_LAND, {nearestAirport: nearestAirport?.airport}, this.aircraft[i]);
+                }
+                PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) is ${aircraft.onGround ? "on ground" : "in the air"} at ${aircraft.latitude}, ${aircraft.longitude} with altitude ${aircraft.barometricAltitude}`);
+
+                this.aircraft[i].meta.liveTrack = true;
+                this.aircraft[i].meta.lastSeen = new Date().getTime();
+                this.aircraft[i].meta.onGround = aircraft.onGround;
+                this.aircraft[i].meta.lat = aircraft.latitude;
+                this.aircraft[i].meta.lon = aircraft.longitude;
+                this.aircraft[i].meta.alt = aircraft.barometricAltitude;
+                this.aircraft[i].meta.squawk = aircraft.squawk;
+            }
+
+            this.save();
+            // PlaneAlert.log.debug(`Plane ${this.name} (${this.icao}) returned data: ${JSON.stringify(data)}`);
+        }
+    }
+
+    private findNearestAirport(aircraft: AircraftMeta) {
+        if (aircraft.meta.lon === null || aircraft.meta.lat === null || PlaneAlert.airports === null) {
+            return null;
+        }
+        PlaneAlert.log.debug(`Plane ${this.name} (${aircraft.icao}) searching for nearest airport of ${aircraft.meta.lat}/${aircraft.meta.lon}`);
+        let min_distance = Number.MAX_SAFE_INTEGER;
+        let nearest_airport = null;
+        for (const airport of PlaneAlert.airports) {
+            if (airport.type === 'closed') {
+                continue;
+            }
+            if (this.allowedAirports.indexOf(airport.type) === -1) {
+                continue;
+            }
+            const distance = GeoUtils.distanceBetweenCoordinates(aircraft.meta.lat, aircraft.meta.lon, airport.latitude_deg, airport.longitude_deg);
+            if (distance < min_distance) {
+                min_distance = distance;
+                nearest_airport = airport;
+            }
+        }
+        return {
+            airport: nearest_airport,
+            distance: min_distance
+        };
+    }
+
+    private async triggerEvent(event: PlaneEvents, data: any = null, aircraft: AircraftMeta) {
```

Comment:
Since there is a new class now (not just Aircraft), this probably should be offloaded to an external helper class

Reply:
Yeah, for sure.